personalities/sctfe/storage/bbolt/dedup.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2024 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bbolt implements modules/dedup using BBolt.
+package bbolt
```

Comment:
Would be nice to expand this package comment with an overview of how it works.

Reply:
Done. There's also a TODO to upload the full design doc under modules/dedup once we've converged on an implementation.